src/tool/python/code_writers.h

```diff
@@ -97,106 +97,84 @@ namespace xlang
         }
     }
 
-    void write_getset_table_prop_row(writer& w, Property const& prop)
-    {
-        auto property_methods = get_property_methods(prop);
-
-        auto getter = w.write_temp("(getter)@_%", prop.Parent().TypeName(), property_methods.get.Name());
-        auto setter = property_methods.set
-            ? w.write_temp("(setter)@_%", prop.Parent().TypeName(), property_methods.set.Name())
-            : w.write_temp("nullptr");
-
-        // TODO: remove const_cast once pywinrt is updated to target Python 3.7. 
-        //       pywinrt currently targeting 3.6 because that's the version that ships with VS 2017 v15.8
-        //       https://github.com/python/cpython/commit/007d7ff73f4e6e65cfafd512f9c23f7b7119b803
-        w.write("    { const_cast<char*>(\"%\"), %, %, nullptr, nullptr },\n",
-            prop.Name(),
-            getter,
-            setter);
-    }
-
-    void write_getset_table_field_row(writer& w, Field const& field)
-    {
-        // TODO: remove const_cast once pywinrt is updated to target Python 3.7. 
-        //       pywinrt currently targeting 3.6 because that's the version that ships with VS 2017 v15.8
-        //       https://github.com/python/cpython/commit/007d7ff73f4e6e65cfafd512f9c23f7b7119b803
-        w.write("    { const_cast<char*>(\"%\"), (getter)@_get_%, (setter)@_set_%, nullptr, nullptr },\n",
-            field.Name(),
-            field.Parent().TypeName(), field.Name(),
-            field.Parent().TypeName(), field.Name());
-    }
-
-    // TODO WinRT Event support
-    //void write_getset_table_row_event(writer& w, Event const& event)
-    //{
-    //    // TODO: remove const_cast once pywinrt is updated to target Python 3.7. 
-    //    //       pywinrt currently targeting 3.6 because that's the version that ships with VS 2017 v15.8
-    //    //       https://github.com/python/cpython/commit/007d7ff73f4e6e65cfafd512f9c23f7b7119b803
-    //    w.write("    { const_cast<char*>(\"%\"), (getter)@_event_%, nullptr, nullptr, nullptr },\n", event.Name(),
-    //        event.Parent().TypeName(), event.Name());
-    //}
-
     void write_getset_table(writer& w, TypeDef const& type)
     {
-        if (has_getsets(type))
-        {
-            w.write("\nstatic PyGetSetDef @_getset[] = {\n", type.TypeName());
+        XLANG_ASSERT(get_category(type) == category::struct_type);
 
-            w.write_each<write_getset_table_prop_row>(get_instance_properties(type));
-            //w.write_each<write_class_getset_table_row_event>(get_instance_events(type));
+        w.write("\nstatic PyGetSetDef @_getset[] = {\n", type.TypeName());
 
-            w.write_each< write_getset_table_field_row>(type.FieldList());
-            w.write("    { nullptr }\n};\n");
+        for (auto&& field : type.FieldList())
+        {
+            // TODO: remove const_cast once pywinrt is updated to target Python 3.7. 

```

Comment:
Any reason why we wouldn't target both? It may mean generating some ugly `#if` checks (or typedefs), but it's worth it for the broader compatibility.

Reply:
It already compiles for both. Haven't addressed packaging yet so I'll keep generating 3.6 compatible code for now